core/lib/spree/money.rb

```diff
@@ -92,6 +94,22 @@ def as_json(*)
       to_s
     end
 
+    def <=>(other)
+      if !other.respond_to?(:money)
+        raise TypeError, "Can't compare #{other.class} to Spree::Money"
+      end
+      if self.currency != other.currency
+        # By default, ::Money will try to run a conversion on `other.money` and
+        # try a comparison on that. We do not want any currency conversion to
+        # take place so we'll catch this here and raise an error.
+        raise(
+          DifferentCurrencyError,
+          "Can't compare #{self.currency} with #{other.currency}"
+        )
+      end
+      self.money <=> other.money
```

Comment:
I'm not sure which one is better here but in the rest of the file to read attributes we used `@money` instead of `self.money`. Is there any reason to change this here? (I think it also applies to `self.currency -> @currency`). If they are equivalent we should probably be more consistent.